parser-typechecker/src/Unison/Codebase/Branch/Type.hs

```diff
@@ -17,15 +29,17 @@ import Unison.NameSegment (NameSegment)
 import Unison.Reference (Reference)
 import Unison.Referent (Referent)
 import Unison.Util.Relation (Relation)
+import Prelude hiding (head)
 
 -- | A node in the Unison namespace hierarchy
 -- along with its history.
 newtype Branch m = Branch {_history :: UnwrappedBranch m}
   deriving (Eq, Ord)
 
-type UnwrappedBranch m = Causal m Raw (Branch0 m)
+type UnwrappedBranch m = Causal m (Branch0 m)
 
-type Hash = Causal.RawHash Raw
+-- | A Hash for a namespace itself, it doesn't incorporate any history.
+type NamespaceHash m = Hash.HashFor (Branch0 m)
```

Comment:
It's admittedly a bit silly to have the `m` here, but this was the easiest way to do this for now, and it wasn't very difficult to plumb things through 🤷🏼

Reply:
Could it go away with a `forall` or something?

Reply:
Unfortunately not, since it would still need to unify with the `m` in scope when you create it from the Causal,  and that concrete `m` won't match `(forall m. m)`.